cunumeric/deferred.py

```diff
@@ -437,20 +451,40 @@ def _create_indexing_array(self, key, is_set=False):
                 N = rhs.ndim
                 out_dtype = rhs.runtime.get_point_type(N)
 
-            out = rhs.runtime.create_unbound_thunk(out_dtype)
+            # TODO : current implementation of the ND output regions
+            # requires out.ndim == rhs.ndim. This will be fixed in the
+            # future
+            out = rhs.runtime.create_unbound_thunk(out_dtype, ndim=rhs.ndim)
+
             task = rhs.context.create_task(CuNumericOpCode.ADVANCED_INDEXING)
             task.add_output(out.base)
             task.add_input(rhs.base)
-            task.add_input(key.base)
+            task.add_input(key)
             task.add_scalar_arg(is_set, bool)
-            task.add_alignment(rhs.base, key.base)
-            task.add_broadcast(
-                key.base, axes=tuple(range(1, len(key.base.shape)))
-            )
+            task.add_scalar_arg(key_dims, ty.int64)
+            task.add_alignment(rhs.base, key)
+            task.add_broadcast(key, axes=tuple(range(1, len(key.shape))))
```

Comment:
this is unnecessary as you already requested alignment between `rhs.base` and `key`.